packages/@adobe/spectrum-css-temp/components/textfield/skin.css

```diff
@@ -28,10 +28,6 @@ governing permissions and limitations under the License.
     &.is-placeholder {
       color: var(--spectrum-textfield-placeholder-text-color-hover);
     }
-
-    & ~ .spectrum-Textfield-icon {
-      fill: var(--spectrum-textfield-icon-color-hover);
-    }
```

Comment:
XD file for searchfield shows that the search icon shouldn't change color unless it is disabled.